extensions/SharkPool/SoundCloud-API.js

```diff
@@ -0,0 +1,363 @@
+              }
+            }
+          },
+        ],
+        menus: {
+          IDS: ["track", "artist"],
+          COMMENT: ["new", "old"],
+          TRACKS: {
+            acceptReporters: true,
+            items: [
+              { text: "name", value: `["title"]` },
+              { text: "artist", value: `["user", "username"]` },
+              { text: "artist ID", value: `["user_id"]` },
+              { text: "description", value: `["description"]` },
+              { text: "cover", value: `["artwork_url"]` },
+              { text: "release date", value: `["created_at"]` },
+              { text: "duration", value: `["duration"]` },
+              { text: "plays", value: `["playback_count"]` },
+              { text: "likes", value: `["likes_count"]` },
+              { text: "comment count", value: `["comment_count"]` },
+              { text: "genre", value: `["genre"]` }
+            ]
+          },
+          ARTISTS: {
+            acceptReporters: true,
+            items: [
+              { text: "username", value: "username" },
+              { text: "description", value: "description" },
+              { text: "profile picture", value: "avatar_url" },
+              { text: "join date", value: "created_at" },
+              { text: "track count", value: "track_count" },
+              { text: "follower count", value: "followers_count" },
+              { text: "following count", value: "followings_count" }
+            ]
+          },
+        },
+      };
+    }
+
+    setClient(args) { clientID = args.ID }
+
+    getClientID() { return clientID }
+
+    async extractID(args) {
+      try {
+        let response;
+        if (!args.URL.startsWith("https://soundcloud")) return "";
+        if (!await Scratch.canFetch(args.URL)) return "";
+        let success = false;
+        let proxyUse = proxy;
+        while (!success && proxyUse) {
+          try {
+            // eslint-disable-next-line
+            response = await fetch(`${proxyUse}${args.URL}`);
+            if (!response.ok) {
+              if (proxyUse === proxy) proxyUse = proxy2;
+              else return "Error: 404";
+            } else { success = true }
+          } catch {
+            if (proxyUse === proxy) proxyUse = proxy2;
+            else return "";
+          }
+        }
+        if (success) {
+          response = await response.text();
+          const regex = args.THING === "track" ? /soundcloud:\/\/sounds:(\d+)/ : /soundcloud:\/\/users:(\d+)/;
+          const match = response.match(regex);
+          if (match && match[1]) return match[1];
+        }
+      } catch { return "" }
+      return "";
+    }
+
+    async getTrackAtt(args) {
+      try {
+        let response;
+        const path = JSON.parse(args.THING);
+        if (!Array.isArray(path)) return "";
+        if (!await Scratch.canFetch(`https://api-v2.soundcloud.com/tracks/${args.ID}?client_id=${clientID}&app_locale=en`)) return "";
+        // eslint-disable-next-line
```

Comment:
Use `Scratch.fetch` or write a comment why you can't. It's important to know for maintenance.